ids/node_id.go

```diff
@@ -58,35 +52,45 @@ func (id *NodeID) UnmarshalJSON(b []byte) error {
 	}
 
 	var err error
-	*id, err = NodeIDFromString(str[1:lastIndex])
+	*nodeID, err = NodeIDFromString(str[1:lastIndex])
 	return err
 }
 
-func (id *NodeID) UnmarshalText(text []byte) error {
-	return id.UnmarshalJSON(text)
+func (nodeID NodeID) MarshalText() ([]byte, error) {
+	return []byte(nodeID.String()), nil
+}
+
+func (nodeID *NodeID) UnmarshalText(text []byte) error {
+	return nodeID.UnmarshalJSON(text)
+}
+
+func (nodeID NodeID) Less(other NodeID) bool {
+	return nodeID < other
 }
 
-func (id NodeID) Less(other NodeID) bool {
-	return bytes.Compare(id[:], other[:]) == -1
+func NodeIDFromBytes(src []byte, length int) NodeID {
+	bytes := make([]byte, length)
+	copy(bytes, src)
+	return NodeID(string(bytes))
```

Comment:
Should we make this a bit more explicitly around being used for tests? I feel like we shouldn't use this in prod code.